### ISSUE TITLE
MD014 only handles the first instance of `$ command` in a group

### ISSUE THREAD

**Issue by TheJaredWilcurt** [DavidAnson/markdownlint#217]
**MD014** only handles first instance of `$ command`. This is inconsistent with how it is documented.

* https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md014---dollar-signs-used-before-commands-without-showing-output

✔️ This throws error:

```sh
$ ls
```

✔️ This does not:

```sh
$ ls
file.md other.md
```

❌ This does not throw an error

```sh
$ ls
file.md other.md
$ git branch
$ cat stuff
```

❌ This does not throw an error

```sh
asdf
$ ls
$ git branch
$ cat stuff
```
